src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1626,15 +1626,15 @@ private void EmitInstanceCallExpression(BoundCall call, UseKind useKind)
                 }
                 else
                 {
-                    // calling a method defined in a base class.
+                    // calling a method defined in a base class or interface.
 
                     // When calling a method that is virtual in metadata on a struct receiver, 
                     // we use a constrained virtual call. If possible, it will skip boxing.
                     if (method.IsMetadataVirtual())
                     {
                         // NB: all methods that a struct could inherit from bases are non-mutating
                         //     treat receiver as ReadOnly
-                        tempOpt = EmitReceiverRef(receiver, AddressKind.ReadOnly);
+                        tempOpt = EmitReceiverRef(receiver, methodContainingType.IsInterface ? AddressKind.Writeable : AddressKind.ReadOnly);
```

Comment:
>tempOpt = EmitReceiverRef(receiver, methodContainingType.IsInterface ? AddressKind.Writeable : AddressKind.ReadOnly); [](http://example.com/codeflow?start=24&length=117)

Does this change also affect readonly fields?

Reply:
Readonly fields are not affected by this change. Here we're only dealing with calls to a method.

Reply:
Did some investigation. It is indeed possible to hit this with a readonly field as well.

On the VB side, it's a bit confusing because the `AddressKind` enum values have different meaning than the C# `AddressKind`s of the same name, but as far as I could tell we didn't have the same problem. Added tests